### PR TITLE
Adding momentum body force for user defined region.

### PR DESCRIPTION
**Pull request by marchdf**
**Pull-request type:**
- [ ] Bug fix 
- [ ] Documentation update
- [X] Feature enhancement

## Description of the pull-request

This PR adds a momentum body force source term for a user defined region.

## Checklist

**NOTE** The checklist below is a suggestion and not mandatory. You don't have
to _check all boxes_ to submit a pull request. However, please add a brief
explanation in your pull request summary explaining the omission for the benefit
of reviewers and developers.

*All pull requests*
- [X] Builds successfully (*must test on at least one system/compiler combination*)
  - Operating systems 
    - [X] Linux
    - [X] MacOS
  - Compilers 
    - [X] GCC
    - [X] LLVM/Clang
    - [X] Intel compilers
    - [ ] NVIDIA CUDA
- [X] Compiles without warnings
- [X] Passes all unit tests
- [X] Passes all regression tests
- [ ] Documentation builds without errors

*For new code updates*
- [X] Documentation updates - additions to user, theory, and verification manuals
- [X] New unit tests providing coverage for new code, bug fixes etc.
- [X] New regression tests exercising the new code

### CODE REVIEW

File: src/node_kernels/MomentumBodyForceBoxNodeKernel.C [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  NodeKernelTraits::LhsType&,
+  NodeKernelTraits::RhsType& rhs,
+  const stk::mesh::FastMeshIndex& node)
+{
+
+  bool is_inside = true;
+  for (int i = 0; i < nDim_; ++i)
+    is_inside =
+      (is_inside && (lo_[i] <= coordinates_.get(node, i)) &&
+       (coordinates_.get(node, i) <= hi_[i]));
```
Comment (sayerhs) [inline src/node_kernels/MomentumBodyForceBoxNodeKernel.C:59]:
```suggestion
  bool is_inside = true;
  for (int i = 0; i < nDim_; ++i)
    if (!((lo_[i] <= coordinates_,get(node, i)) && (coordinates_.get(node, i) <= hi_[i])) {
      is_inside = false;
      break;
    }
```

File: src/node_kernels/MomentumBodyForceBoxNodeKernel.C [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+  const NodeKernelTraits::DblType dualVolume = dualNodalVolume_.get(node, 0);
+  for (int i = 0; i < nDim_; ++i)
+    rhs(i) += (is_inside) ? dualVolume * forceVector_[i] : 0.0;
```
Comment (sayerhs) [inline src/node_kernels/MomentumBodyForceBoxNodeKernel.C:63]:
```suggestion
  if (is_inside) {
    const NodeKernelTraits::DblType dualVolume = dualNodalVolume_.get(node, 0);
    for (int i = 0; i < nDim_; ++i)
      rhs(i) += dualVolume * forceVector_[i];
   }
```

File: src/node_kernels/MomentumBodyForceBoxNodeKernel.C (hunk 1 above)
Comment (marchdf) [reply src/node_kernels/MomentumBodyForceBoxNodeKernel.C]: fixed

File: src/node_kernels/MomentumBodyForceBoxNodeKernel.C (hunk 2 above)
Comment (marchdf) [reply src/node_kernels/MomentumBodyForceBoxNodeKernel.C]: fixed

**Pull request merged**